nf_core/sync.py

```diff
@@ -239,60 +243,29 @@ def push_template_branch(self):
-                    self.gh_repo, self.original_branch
-                )
-            )
-
-        log.info("Submitting a pull request via the GitHub API")
-
-        pr_title = "Important! Template update for nf-core/tools v{}".format(nf_core.__version__)
-        pr_body_text = (
-            "A new release of the main template in nf-core/tools has just been released. "
-            "This automated pull-request attempts to apply the relevant updates to this pipeline.\n\n"
-            "Please make sure to merge this pull-request as soon as possible. "
-            "Once complete, make a new minor release of your pipeline. "
-            "For instructions on how to merge this PR, please see "
-            "[https://nf-co.re/developers/sync](https://nf-co.re/developers/sync#merging-automated-prs).\n\n"
-            "For more information about this release of [nf-core/tools](https://github.com/nf-core/tools), "
-            "please see the [nf-core/tools v{tag} release page](https://github.com/nf-core/tools/releases/tag/{tag})."
-        ).format(tag=nf_core.__version__)
-
-        # Try to update an existing pull-request
-        if self.update_existing_pull_request(pr_title, pr_body_text) is False:
-            # None found - make a new pull-request
-            self.submit_pull_request(pr_title, pr_body_text)
-
-    def update_existing_pull_request(self, pr_title, pr_body_text):
-        """
-        List existing pull-requests between TEMPLATE and self.from_branch
-
-        If one is found, attempt to update it with a new title and body text
-        If none are found, return False
+    def close_open_template_merge_pull_requests(self):
+        """Get all template merging branches (starting with 'nf-core-template-merge-')
+        and check for any open PRs from these branches to the self.from_branch
+        If open PRs are found, add a comment and close them
         """
         assert os.environ.get("GITHUB_AUTH_TOKEN", "") != ""
+        log.info("Checking for open PRs from template merge branches")
+        # Get list of all branches
+        branch_list = [b.name for b in self.repo.branches]
+        # Subset to template merging branches
+        branch_list = [b for b in branch_list if b.startswith("nf-core-template-merge-")]
+        for branch in branch_list:
+            # Check for open PRs and close if found
+            self.close_open_pr(branch)
+
+    def close_open_pr(self, branch):
+        """Given a branch, check for open PRs from that branch to self.from_branch
+        and close if PRs have been found
+        """
+        log.info("Checking branch: {}".format(branch))
         # Look for existing pull-requests
-        list_prs_url = "https://api.github.com/repos/{}/pulls?head=nf-core:TEMPLATE&base={}".format(
-            self.gh_repo, self.from_branch
+        list_prs_url = "https://api.github.com/repos/{}/pulls?head={}&base={}".format(
```

Comment:
This might be a good time to get started with f-strings in tools. Much more readable than this.

Reply:
I always use f-strings for everything else I do in Python - a thousand times better. So I would be happy to start using them - question is only how bad it is in terms of style to use them only here and there (because it won't change overnight). Something for @ewels to decide :)

Reply:
Ah sorry, only just saw this. Yes use them - I initially avoided them as in the early days of this package a few years ago we were trying to maintain compatibility with Py 2k. But we dropped that a while back and since then I've been putting stuff into f-strings as I come across them (where it makes sense to do so anyway).